File: examples/python3/chat-log.py
# Chat Log
#
# 1. se connecte en mqtt à `localhost` (port 1883)
# 2. souscrit au sujet (topic) `chat_messages`
# 3. lorsqu'un message est reçu, si il est du sujet `chat_messages`, l'afficher
#    sur la sortie standard

import paho.mqtt.client as mqtt
broker_host = 'localhost'
topic_string = 'chat_messages'


def on_connect(client, userdata, flags, rc):
    print('connecté !')
    print('souscription à %s ...' % repr(topic_string))
    client.subscribe(topic_string)


def on_message(client, userdata, msg):
    if msg.topic == topic_string:
        payload = str(msg.payload, encoding='utf8')
        print('message reçu: %s (%s)' % (repr(payload), msg.topic))


if __name__ == '__main__':
    client = mqtt.Client()
    client.on_connect = on_connect
    client.on_message = on_message
    client.connect(broker_host, 1883, 60)
    client.loop_forever()

File: examples/nodejs/chat-log.js
// # Chat Log
//
// 1. se connecte en mqtt à `localhost` (port 1883)
// 2. souscrit au sujet (topic) `chat_messages`
// 3. lorsqu'un message est reçu, si il est du sujet `chat_messages`, l'afficher
//    sur la sortie standard

let mqtt = require('mqtt')
let broker_uri = 'mqtt://localhost'
let topic_string = 'chat_messages'

console.log('connexion à %j ...', broker_uri)
let client = mqtt.connect(broker_uri)

client.on('connect', function () {
  console.log('connecté !')
  console.log('souscription à %j ...', topic_string)
  client.subscribe(topic_string, function (err) {
    if (err) {
      console.error(err)
    }
  })
})

client.on('message', function (topic, message) {
  // message contient le contenu du message est un buffer
  if (topic === topic_string) {
    console.log("message reçu: %j (%s)", message.toString(), topic)
  }
})
